vm_insnhelper.c

```diff
@@ -4926,40 +4926,81 @@ vm_opt_newarray_min(rb_execution_context_t *ec, rb_num_t num, const VALUE *ptr)
 
 #define IMEMO_CONST_CACHE_SHAREABLE IMEMO_FL_USER0
 
-// For each getconstant, associate the ID that corresponds to the first operand
-// to that instruction with the inline cache.
+// This function associates a given ID that is part of a constant reference with
+// a given inline cache. It is used as a callback when walking through ISEQs
+// whenever a getconstant instruction is found.
+static void
+vm_ic_compile_i(ID id, void *ic)
+{
+    rb_vm_t *vm = GET_VM();
+    st_table *ics;
+
+    if (!rb_id_table_lookup(vm->constant_cache, id, (VALUE *) &ics)) {
+        ics = st_init_numtable();
+        rb_id_table_insert(vm->constant_cache, id, (VALUE) ics);
+    }
+
+    st_insert(ics, (st_data_t) ic, (st_data_t) Qtrue);
+}
+
+// This struct is used to pass data between the rb_vm_iseq_each_getconstant_id
+// function and the vm_iseq_each_getconstant_id_i function that it is using as
+// an iterator. The callback will be called for each ID that is associated with
+// a getconstant instruction. It will be passed both the ID and whatever value
+// is stored in the void* pointer.
+struct rb_vm_iseq_each_getconstant_id_data {
+    void (*callback)(ID, void*);
+    void *argument;
+};
+
+// This function is used as a callback to the ISEQ-walking function
+// rb_iseq_each. It is used to find all of the getconstant instructions between
+// a pair of opt_getinlinecache/opt_setinlinecache instructions for the purpose
+// of associating those IDs with inline caches.
+//
+// It is does in a generic way here with a callback function that is passed in
+// through the data pointer so that it can be reused by YJIT when tracking which
+// constants are associated with particular blocks.
 static bool
-vm_ic_compile_i(VALUE *code, VALUE insn, size_t index, void *ic)
+vm_iseq_each_getconstant_id_i(VALUE *code, VALUE insn, size_t index, void *data)
 {
     if (insn == BIN(opt_setinlinecache)) {
         return false;
     }
 
     if (insn == BIN(getconstant)) {
-        ID id = code[index + 1];
-        rb_vm_t *vm = GET_VM();
-
-        st_table *ics;
-        if (!rb_id_table_lookup(vm->constant_cache, id, (VALUE *) &ics)) {
-            ics = st_init_numtable();
-            rb_id_table_insert(vm->constant_cache, id, (VALUE) ics);
-        }
-
-        st_insert(ics, (st_data_t) ic, (st_data_t) Qtrue);
+        struct rb_vm_iseq_each_getconstant_id_data *d = data;
+        d->callback(code[index + 1], d->argument);
     }
 
     return true;
 }
 
+// This function is used to trigger a walk through an ISEQ using rb_iseq_each to
+// find all of the IDs associated with the getconstant instructions within a
+// certain range of the body of the ISEQ. It is used to both associate those IDs
+// with inline caches (in the case of vm_ic_compile) as well as to track those
+// IDs with their associated YJIT blocks.
+void
```

Comment:
```suggestion
MJIT_STATIC void
```

I think this should fix the MJIT Windows CI